src/libcore/atomic.rs

```diff
@@ -1089,3 +1090,21 @@ pub fn fence(order: Ordering) {
         }
     }
 }
+
+macro_rules! impl_Debug {
+    ($($t:ident)*) => ($(
+        impl fmt::Debug for $t {
```

Comment:
Does it though? Either you have a compiler that can build it, or you don't. I suppose if you have one that can build it, you can query if you can use an older one...?